libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationLocationEngineListenerTest.java

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+
+import com.mapbox.android.core.location.LocationEngine;
+
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class NavigationLocationEngineListenerTest {
+
+  @Test
+  public void onConnected_engineRequestsUpdates() {
+    LocationEngine locationEngine = mock(LocationEngine.class);
+    NavigationLocationEngineListener listener = buildListener(locationEngine);
+
+    listener.onConnected();
+
+    verify(locationEngine).requestLocationUpdates();
+  }
+
+  @Test
+  public void queueLocationUpdate_threadReceivesUpdate() {
+    RouteProcessorBackgroundThread thread = mock(RouteProcessorBackgroundThread.class);
+    NavigationLocationEngineListener listener = buildListener(thread);
+
+    listener.queueLocationUpdate(mock(Location.class));
```

Comment:
Why not testing this through the `public` API `onLocationChanged`? That way we'd be able to test the unhappy path (not a valid location update) as well?